pkg/cli/admin/inspect/resource.go

```diff
@@ -274,15 +280,23 @@ func obtainRelatedObjects(obj *unstructured.Unstructured) ([]*configv1.ObjectRef
 		return nil, nil
 	}
 
+	errs := []error{}
 	relatedObjs := []*configv1.ObjectReference{}
-	for _, relatedObj := range val {
+	for i, relatedObj := range val {
 		ref := &configv1.ObjectReference{}
 		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(relatedObj.(map[string]interface{}), ref); err != nil {
 			return nil, err
 		}
-		relatedObjs = append(relatedObjs, ref)
-		klog.V(1).Infof("    Found related object %q...\n", objectReferenceToString(ref))
+
+		refStr := objectReferenceToString(ref)
+		if refStr != "" {
+			klog.V(1).Infof("    Found related object %q...\n", refStr)
+			relatedObjs = append(relatedObjs, ref)
+		} else {
+			errs = append(errs, fmt.Errorf("skipping .status.relatedObjects[%v] in %q because the ObjectReference is invalid",
+				i, unstructuredToString(obj)))
```

Comment:
If an error is not generated here, and the invalid related object is just skipped instead, then the command exits cleanly. Personally, I think that would be a bad change in behavior.

```
❯ ./oc adm inspect configurationpolicies.policy.open-cluster-management.io --all-namespaces --v=1
I1027 11:31:04.951457   20922 resource.go:275] Gathering related object reference information for "ConfigurationPolicy/policy-namespace-jkuli"...
I1027 11:31:04.953215   20922 resource.go:275] Gathering related object reference information for "ConfigurationPolicy/policy-pod-jkuli"...
Wrote inspect data to inspect.local.8438621565858530944.

❯ echo $?
0

❯ tree inspect.local.8438621565858530944 -h
[ 160]  inspect.local.8438621565858530944
├── [3.3K]  event-filter.html
├── [  96]  namespaces
│   └── [  96]  local-cluster
│       └── [  96]  policy.open-cluster-management.io
│           └── [ 128]  configurationpolicies
│               ├── [4.0K]  policy-namespace-jkuli.yaml
│               └── [3.7K]  policy-pod-jkuli.yaml
└── [ 104]  timestamp

5 directories, 4 files
```